### PR TITLE
Fix bugs relating to profiles during `test-all`

### PR DESCRIPTION
**Pull request by aengelberg**
- :jvm-opts within :profiles were not being properly merged into the
  original project
- test-all was not respecting :test profile

### CODE REVIEW

File: external/lein-repo/src/lein_repo/plugin.clj [hunk 1]
```diff
@@ -4,7 +4,7 @@
             [clojure.set :as set]
             [clojure.pprint :as pprint]
             [clojure.java.shell :as shell]
-            leiningen.core.project)
+            [leiningen.core.project :as project]
```
Comment (greglook) [inline external/lein-repo/src/lein_repo/plugin.clj]: Looks like you're missing the closing paren for the `:require` form.

File: external/lein-repo/src/lein_repo/plugin.clj [hunk 2]
```diff
@@ -124,11 +124,13 @@
...
    :repl-options (fn [p1 p2] (merge-with concat-distinct p1 p2))
-   :profiles (fn [p1 p2] (merge-with merge-projects p1 p2))
+   :profiles (fn [p1 p2]
+               (merge-with merge-projects p1 p2))
 
    ;; cljx specific
    :prep-tasks concat-distinct
-   :cljx (fn [a b] {:builds (concat-distinct (:builds a) (:builds b))})})
+   :cljx (fn [a b] {:builds (concat-distinct (:builds a) (:builds b))})
+   :jvm-opts concat-distinct})
```
Comment (greglook) [inline external/lein-repo/src/lein_repo/plugin.clj]: Maybe put this under the `:profiles` line above? As written it looks like the "cljx specific" comment applies to this too.

File: external/lein-repo/src/lein_repo/plugin.clj [hunk 3]
```diff
@@ -223,6 +228,11 @@
                 :eval-in :subprocess
                 :jvm-opts ^:replace ["-Xmx600m" "-XX:MaxPermSize=256m"])))))
 
+(defn merge-test-profile
+  "Can't figure out how to make the task respect the :test profile, so
+  just merge the :test profile map into the project"
```
Comment (greglook) [inline external/lein-repo/src/lein_repo/plugin.clj:233]: There's a better way to do this, but I don't think it's worth refactoring the plugin enough to modernize it.

File: external/lein-repo/src/lein_repo/plugin.clj [hunk 4]
```diff
@@ -234,18 +244,18 @@
...
 
-(def test-all-project
-  (delay (test-all-project*
-          (read-project-file (first (all-internal-deps))))))
+(def test-all-project (memoize test-all-project*))
 
-(defn ordered-source-and-test-paths [& [no-tests?]]
-  (let [mega-project @test-all-project
+(defn ordered-source-and-test-paths [base & [no-tests?]]
+  (let [mega-project (test-all-project base)
```
Comment (greglook) [inline external/lein-repo/src/lein_repo/plugin.clj:258]: Nice, I was wondering why the code was written this way (not using the project map from the actual project!) anyway.

**Comment by greglook** [comment]
LGTM :shipit:

File: external/lein-repo/src/lein_repo/plugin.clj (hunk 3 above)
Comment (aengelberg) [inline external/lein-repo/src/lein_repo/plugin.clj:233]: I spent some effort getting it to work via `(leiningen.core.project/merge-profiles project [:test :leiningen/test])` like `lein test` does. That helper function apparently tries to do a bunch of other stuff I don't want.

File: external/lein-repo/src/lein_repo/plugin.clj (hunk 1 above)
Comment (aengelberg) [inline external/lein-repo/src/lein_repo/plugin.clj]: Yeesh! I forgot to try rebuilding it after I removed a bunch of debugging stuff. Thanks for pointing that out.

**Pull request merged**